### PR TITLE
Update pointcloud.py to accept NeumannBC

### PR DESCRIPTION
**Pull request by mitchelldaneker**
Added `boundary_normal(self,x)` function so NeumannBC can be used. Has been tested on simple 1D case and same solution was found for self defined normals.

### CODE REVIEW

File: deepxde/geometry/pointcloud.py [hunk 1]
```diff
@@ -68,6 +68,13 @@ def on_boundary(self, x):
             .any(axis=1)
         )
 
+    def boundary_normal(self, x):
+        if self.boundary_normals is None:
+            raise ValueError(
+                "boundary_normals must be defined for boundary_normal"
+            )
+        return self.boundary_normals
```
Comment (lululxvi) [inline deepxde/geometry/pointcloud.py:76]: Should only return the normal for those x in the function argument.

Comment (lululxvi) [reply deepxde/geometry/pointcloud.py:70]: @mitchelldaneker

Comment (mitchelldaneker) [reply deepxde/geometry/pointcloud.py:70]: I need to determine a way to find those x in the function argument. So far, the only methods I have been able to create are slow, and even when using matrix alg. are not fast enough to work for when there are many inputs.

Comment (mitchelldaneker) [reply deepxde/geometry/pointcloud.py]: I have implemented a method similar to on_boundary. This uses np.where, and allows for repeats in the x input.

File: deepxde/geometry/pointcloud.py [hunk 2]
```diff
@@ -62,6 +62,17 @@ def on_boundary(self, x):
             .any(axis=1)
         )
 
+    def boundary_normal(self, x):
+        if self.boundary_normals is None:
+            raise ValueError(
+                "boundary_normals must be defined for boundary_normal"
+            )
+        boundary_point_matches = np.isclose(
```
Comment (lululxvi) [inline deepxde/geometry/pointcloud.py:70]: Replace `np.isclose` with `isclose`

**Comment by lululxvi** [comment]
Please fix the Codacy error.

**Comment by mitchelldaneker** [comment]
> Please fix the Codacy error.

The current implementation is correct as per numpy documentation. Suggestions are incorrect as well.

**Comment by lululxvi** [comment]
> > Please fix the Codacy error.
> 
> The current implementation is correct as per numpy documentation. Suggestions by Codacy are incorrect as well.

Are you sure? References?

**Comment by mitchelldaneker** [comment]
> > > Please fix the Codacy error.
> > 
> > 
> > The current implementation is correct as per numpy documentation. Suggestions by Codacy are incorrect as well.
> 
> Are you sure? References?

Removed a redundancy which should fix the error. It seemed codacy did not know what I was trying to do, so gave incorrect suggestions that broke the code. Since it is already a boolean array, do not need `== True`.

**Pull request merged**